runtime/src/main/java/org/corfudb/protocols/wireprotocol/NettyCorfuMessageDecoder.java

```diff
@@ -19,6 +20,13 @@
     @Override
     protected void decode(ChannelHandlerContext channelHandlerContext, ByteBuf byteBuf,
                           List<Object> list) throws Exception {
+        // Leave message unchanged if is non-legacy
+        if(byteBuf.getByte(byteBuf.readerIndex()) != API.LEGACY_CORFU_MSG_MARK) {
```

Comment:
Never mind, ignore my comment below. It's wrong.

Unline the readByte, getByte API doesn't move the reader index.